### ISSUE TITLE
React Native - gradle, manifest merge error

### ISSUE THREAD

**Issue by ratz6** [facebook/react-native#32657]
### Description

Execution failed for task ':app:processReleaseMainManifest'.
> Manifest merger failed : Attribute meta-data#org.unimodules.core.AppLoader#react-native-headless@value value=(org.unimodules.adapters.react.apploader.RNHeadlessAppLoader) from [:unimodules-react-native-adapter] AndroidManifest.xml:14:13-88
        is also present at [:expo-modules-core] AndroidManifest.xml:14:13-86 value=(expo.modules.adapters.react.apploader.RNHeadlessAppLoader).
        Suggestion: add 'tools:replace="android:value"' to <meta-data> element at AndroidManifest.xml:12:9-14:91 to override.

### Version

0.63.0

### Output of `react-native info`

System:
    OS: Linux 5.11 Ubuntu 20.04.2 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz
    Memory: 1.15 GB / 30.97 GB
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 14.18.1 - /usr/bin/node
    Yarn: 1.22.11 - /usr/bin/yarn
    npm: 6.14.15 - /usr/bin/npm
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
  Languages:
    Java: 11.0.11 - /usr/bin/javac
    Python: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: 0.63.0 => 0.63.0 
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

I cloned my code and install the node modules using yarn install, then start it with 'yarn start' and finally run with 'yarn run android'.
It used to work since 4 months and surprisingly yesterday all of a sudden I start getting the error while performing yarn run android.
All things in react native cli. 

### Snack, code example, screenshot, or link to a repository

https://stackoverflow.com/questions/69810291/react-native-gradle-manifest-merge-error

**Comment by RRajnishR** [comment]
Facing the Same Issue, were you able to fix this issue? It happened suddenly, the application was working fine till today afternoon.

**Comment by ratz6** [comment]
My dev is stuck on this for 4 days. I tried deleting all the project dependencies and still it didn't work so I went ahead and formatted my OS, but still I get the error :(
Let me know in case you come across a fix

**Comment by cortinico** [comment]
That's essentially a duplicate of #32460 The resolution provided there should fix also this issue.

**Comment by ratz6** [comment]
@cortinico But I don't see a fix there for react native cli, they were talking about upgrading expo version which worked for them

**Comment by RRajnishR** [comment]
> @cortinico But I don't see a fix there for react native cli, they were talking about upgrading expo version which worked for them

You did not have to format your OS. @cortinico has provided a solution [Here](https://github.com/facebook/react-native/issues/32460#issuecomment-955732630). Although, I fixed it a different way.

This error occurred in my **expo-ejected** application when I ran "./gradlew clean" command after doing some changes. At first, the error was of "node_modules/@unimodules" was not found, so I removed node_modules + package-lock.json then performed a "npm install". Which In turn gave me the afore-mentioned error. after 2 hours of searching and head-scratching. I went back to the state where I first encountered the unimodules error(ofcourse because of git) and just installed "react-native-unimodules" package seperately. It seems to have fixed the issue for now.

**Comment by ratz6** [comment]
@cortinico has asked to do to avoid the mis-match of both modules and after doing that I get a new error as mentioned in the links of stackoverflow for the same issue ...

 Error - error: method asList in class Arrays cannot be applied to given types;
    return Arrays.<Package>asList(
                 ^
  required: T[]
  found: SplashScreenPackage,AVPackage,CameraPackage,ConstantsPackage,DocumentPickerPackage,ErrorRecoveryPackage,FileSystemPackage,FontLoaderPackage,ImageLoaderPackage,PermissionsPackage,ImagePickerPackage,KeepAwakePackage,LinearGradientPackage,SecureStorePackage,SQLitePackage,UpdatesPackage,WebBrowserPackage,ApplicationPackage,LocationPackage
  reason: varargs mismatch; SplashScreenPackage cannot be converted to Package
  where T is a type-variable:
    T extends Object declared in method <T>asList(T...)
    
    
    My Android Manifest file : > 
    
    <manifest xmlns:android="http://schemas.android.com/apk/res/android" xmlns:tools="http://schemas.android.com/tools" package="live.streamstyle">
  <uses-permission android:name="android.permission.INTERNET"/>
  <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
  <uses-permission android:name="android.permission.READ_PHONE_STATE"/>
  <uses-permission android:name="android.permission.USE_FINGERPRINT"/>
  <uses-permission android:name="android.permission.USE_BIOMETRIC"/>
  <uses-permission android:name="android.permission.VIBRATE"/>
  <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS"/>
  <uses-permission android:name="android.permission.CAMERA"/>
  <uses-permission android:name="android.permission.RECORD_AUDIO"/>
  <uses-permission android:name="android.permission.READ_CONTACTS"/>
  <uses-permission android:name="android.permission.WRITE_CONTACTS"/>
  <uses-permission android:name="android.permission.READ_CALENDAR"/>
  <uses-permission android:name="android.permission.WRITE_CALENDAR"/>
  <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
  <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
  <uses-permission android:name="android.permission.WRITE_SETTINGS"/>
  <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
  <uses-permission android:name="android.permission.WAKE_LOCK"/>
  <uses-permission android:name="com.google.android.c2dm.permission.RECEIVE"/>

  <uses-permission android:name="android.permission.INTERACT_ACROSS_USERS_FULL" />

  <application android:name=".MainApplication" android:label="@string/app_name" android:icon="@mipmap/ic_launcher" android:roundIcon="@mipmap/ic_launcher_round" android:allowBackup="true" android:theme="@style/AppTheme" android:usesCleartextTraffic="true" tools:node="replace">
    <meta-data android:name="expo.modules.updates.EXPO_UPDATE_URL" android:value="https://exp.host/@my-expo-username/my-app" /> 
    <meta-data android:name="expo.modules.updates.EXPO_SDK_VERSION" android:value="40.0.0"/>
    <meta-data android:name="expo.modules.updates.ENABLED" android:value="true"/>
    <meta-data android:name="expo.modules.updates.EXPO_UPDATES_CHECK_ON_LAUNCH" android:value="ALWAYS"/>
    <meta-data android:name="expo.modules.updates.EXPO_UPDATES_LAUNCH_WAIT_MS" android:value="0"/>
    <meta-data android:name="org.unimodules.core.AppLoader#react-native-headless" android:value="expo.modules.adapters.react.apploader.RNHeadlessAppLoader" tools:replace="android:value" />
    <activity android:name=".MainActivity" android:label="@string/app_name" android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode" android:launchMode="singleTask" android:windowSoftInputMode="adjustResize" android:theme="@style/Theme.App.SplashScreen" android:screenOrientation="portrait">
      <intent-filter>
        <action android:name="android.intent.action.MAIN"/>
        <category android:name="android.intent.category.LAUNCHER"/>
      </intent-filter>
      <intent-filter>
        <action android:name="android.intent.action.VIEW"/>
        <category android:name="android.intent.category.DEFAULT"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <data android:scheme="live.streamstyle"/>
      </intent-filter>
      <intent-filter android:autoVerify="true">
        <action android:name="android.intent.action.VIEW" />
        <category android:name="android.intent.category.DEFAULT" />
        <category android:name="android.intent.category.BROWSABLE" />
        
        <data android:scheme="https"
              android:host="${appDomain}"
               />
        
    </intent-filter>
    </activity>
    <activity android:name="com.facebook.react.devsupport.DevSettingsActivity"/>
  </application>
</manifest>

**Comment by ratz6** [comment]
@cortinico has asked to do to avoid the mis-match of both modules and after doing that I get a new error as mentioned in the links of stackoverflow for the same issue ...

 Error - error: method asList in class Arrays cannot be applied to given types;
    return Arrays.<Package>asList(
                 ^
  required: T[]
  found: SplashScreenPackage,AVPackage,CameraPackage,ConstantsPackage,DocumentPickerPackage,ErrorRecoveryPackage,FileSystemPackage,FontLoaderPackage,ImageLoaderPackage,PermissionsPackage,ImagePickerPackage,KeepAwakePackage,LinearGradientPackage,SecureStorePackage,SQLitePackage,UpdatesPackage,WebBrowserPackage,ApplicationPackage,LocationPackage
  reason: varargs mismatch; SplashScreenPackage cannot be converted to Package
  where T is a type-variable:
    T extends Object declared in method <T>asList(T...)

**Comment by ratz6** [comment]
@RRajnishR I get an error which I do a npm run-android or yarn run android . When I do ./gradlew clean I don't see any errors. So installing uni-modules separately worked for you ?

**Comment by ratz6** [comment]
BasePackageList.java > 
`package live.streamstyle.generated;

import java.util.Arrays;
import java.util.List;
import org.unimodules.core.interfaces.Package;

public class BasePackageList {
  public List<Package> getPackageList() {
    return Arrays.<Package>asList(
        new expo.modules.splashscreen.SplashScreenPackage(),
        new expo.modules.av.AVPackage(),
        new expo.modules.camera.CameraPackage(),
        new expo.modules.constants.ConstantsPackage(),
        new expo.modules.documentpicker.DocumentPickerPackage(),
        new expo.modules.errorrecovery.ErrorRecoveryPackage(),
        new expo.modules.filesystem.FileSystemPackage(),
        new expo.modules.font.FontLoaderPackage(),
        new expo.modules.imageloader.ImageLoaderPackage(),
        new expo.modules.permissions.PermissionsPackage(),
        new expo.modules.imagepicker.ImagePickerPackage(),
        new expo.modules.keepawake.KeepAwakePackage(),
        new expo.modules.lineargradient.LinearGradientPackage(),
        new expo.modules.securestore.SecureStorePackage(),
        new expo.modules.sqlite.SQLitePackage(),
        new expo.modules.updates.UpdatesPackage(),
        new expo.modules.webbrowser.WebBrowserPackage(),
        new expo.modules.application.ApplicationPackage(),
        new expo.modules.location.LocationPackage()
    );
  }
}
`

**Comment by ratz6** [comment]
My Android Manifest (main) >> 

![Screenshot from 2021-11-30 06-44-57](https://user-images.githubusercontent.com/77700933/143967931-c2e6408f-dd89-4b52-abe6-c94586207a9b.png)

**Comment by ratz6** [comment]
My BasePackageList.java >>> 
![Screenshot from 2021-11-30 06-48-31](https://user-images.githubusercontent.com/77700933/143968248-7a617a9b-8d3c-4aaf-9e0f-5f1629e659c7.png)

**Comment by ratz6** [comment]
> You did not have to format your OS. @cortinico has provided a solution [Here](https://github.com/facebook/react-native/issues/32460#issuecomment-955732630). Although, I fixed it a different way.
> 
> This error occurred in my **expo-ejected** application when I ran "./gradlew clean" command after doing some changes. At first, the error was of "node_modules/@unimodules" was not found, so I removed node_modules + package-lock.json then performed a "npm install". Which In turn gave me the afore-mentioned error. after 2 hours of searching and head-scratching. I went back to the state where I first encountered the unimodules error(ofcourse because of git) and just installed "react-native-unimodules" package seperately. It seems to have fixed the issue for now.

@RRajnishR I get an error which I do a npm run-android or yarn run android . When I do ./gradlew clean I don't see any errors. So installing uni-modules separately worked for you ?

**Comment by ratz6** [comment]
> @cortinico has asked to do to avoid the mis-match of both modules and after doing that I get a new error as mentioned in the links of stackoverflow for the same issue ...
> 
> error: method asList in class Arrays cannot be applied to given types; return Arrays.asList( ^ required: T[] found: SplashScreenPackage,AVPackage,CameraPackage,ConstantsPackage,DocumentPickerPackage,ErrorRecoveryPackage,FileSystemPackage,FontLoaderPackage,ImageLoaderPackage,PermissionsPackage,ImagePickerPackage,KeepAwakePackage,LinearGradientPackage,SecureStorePackage,SQLitePackage,UpdatesPackage,WebBrowserPackage,ApplicationPackage,LocationPackage reason: varargs mismatch; SplashScreenPackage cannot be converted to Package where T is a type-variable: T extends Object declared in method asList(T...)

Found out someone else faced the same issue and tried doing what they had done but still getting the same error..
https://stackoverflow.com/questions/69977145/react-native-android-build-fails-with-error-in-basepackagelist-java-varargs-mis

**Comment by RRajnishR** [comment]
> -modules separately worked for you

To give context, I am facing a "node_modules not found" error here and there, after I ejected the expo. So, what I do is reinstall the specific module which seems to be missing. this usually occurs when I have to do a ./gradlew clean. 

You can try updating the expo, but I'll advise you to clone your project to a separate folder and checkout to any old commit and update the new changes

**Comment by RRajnishR** [comment]
> > You did not have to format your OS. @cortinico has provided a solution [Here](https://github.com/facebook/react-native/issues/32460#issuecomment-955732630). Although, I fixed it a different way.
> > This error occurred in my **expo-ejected** application when I ran "./gradlew clean" command after doing some changes. At first, the error was of "node_modules/@unimodules" was not found, so I removed node_modules + package-lock.json then performed a "npm install". Which In turn gave me the afore-mentioned error. after 2 hours of searching and head-scratching. I went back to the state where I first encountered the unimodules error(ofcourse because of git) and just installed "react-native-unimodules" package seperately. It seems to have fixed the issue for now.
> 
> @RRajnishR I get an error which I do a npm run-android or yarn run android . When I do ./gradlew clean I don't see any errors. So installing uni-modules separately worked for you ? Can you give a little bit context ? I see https://docs.expo.dev/bare/installing-unimodules/ and it's asking to make changes in android activities..

Just check your package.json, "react-native-unimodules" should already be there. You just have to install it once again.

**Comment by ratz6** [comment]
> > -modules separately worked for you
> 
> To give context, I am facing a "node_modules not found" error here and there, after I ejected the expo. So, what I do is reinstall the specific module which seems to be missing. this usually occurs when I have to do a ./gradlew clean.
> 
> You can try updating the expo, but I'll advise you to clone your project to a separate folder and checkout to any old commit and update the new changes

> You can try updating the expo, but I'll advise you to clone your project to a separate folder and checkout to any old commit and update the new changes
Did that for my entire last month of commits, same error everywhere...

But after updating will I have to eject again ? I have already ejected and am in RN CLI , so should I delete my android/ios folders and change expo version in package.json and eject again ??
If that's how it's done all the changes made in android/ios and 3rd party sdk's should again be implemented ? (i mean the code should be matching)

**Comment by ratz6** [comment]
> > > You did not have to format your OS. @cortinico has provided a solution [Here](https://github.com/facebook/react-native/issues/32460#issuecomment-955732630). Although, I fixed it a different way.
> > > This error occurred in my **expo-ejected** application when I ran "./gradlew clean" command after doing some changes. At first, the error was of "node_modules/@unimodules" was not found, so I removed node_modules + package-lock.json then performed a "npm install". Which In turn gave me the afore-mentioned error. after 2 hours of searching and head-scratching. I went back to the state where I first encountered the unimodules error(ofcourse because of git) and just installed "react-native-unimodules" package seperately. It seems to have fixed the issue for now.
> > 
> > 
> > @RRajnishR I get an error which I do a npm run-android or yarn run android . When I do ./gradlew clean I don't see any errors. So installing uni-modules separately worked for you ? Can you give a little bit context ? I see https://docs.expo.dev/bare/installing-unimodules/ and it's asking to make changes in android activities..
> 
> Just check your package.json, "react-native-unimodules" should already be there. You just have to install it once again.

Getting the same error only after doing that
.. but If I do what @cortinico  has done `[here](https://github.com/facebook/react-native/issues/32460#issuecomment-955732630)` I don't get that error but a new error  [``](https://github.com/facebook/react-native/issues/32657#issuecomment-982180637).. It's been 5 days already :(

**Comment by ratz6** [comment]
Any update team ?

**Comment by dduranalba** [comment]
> Any update team ?

Hi there. I was facing this same exact issue and it turned out to be a problem with using expo and react-native-unimodules. Expo has upgraded their modules strategy and they no longer use react-native-unimodules. The module system and autolinking implementation now live in the expo package instead. 

So basically you have to migrate from react-native-unimodules to Expo modules.

This is the official guide from Expo to do so: 
https://dev.to/wbroek/migrate-from-react-native-unimodules-to-expo-modules-25c6

Hope this solve your issue as well.

**Comment by cortinico** [comment]
> So basically you have to migrate from react-native-unimodules to Expo modules.

I'm leaning towards closing this issue. As previously mentioned, this is an issue of either Expo modules or react-native-unimodules and their interaction. 

An issue should be opened on their Github issue tracker instead of here, as this problem is not directly related to react-native.  It will also be beneficial for Expo maintainers to have visibility over this issue.
